File: Beecrowd_Parte5/Van_2693.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;
import java.util.Scanner;

// Classe que representa um aluno com nome, cidade e idade
class Aluno {
    String nome;
    String cidade;
    int idade;

    // Construtor
    public Aluno(String nome, String cidade, int idade) {
        this.nome = nome;
        this.cidade = cidade;
        this.idade = idade;
    }

    // Método toString para facilitar a impressão
    @Override
    public String toString() {
        return nome;
    }
}

public class Main {
    // Comparator para ordenação de Alunos
    static class AlunoComparator implements Comparator<Aluno> {
        @Override
        public int compare(Aluno a, Aluno b) {
            if (a.idade != b.idade) {
                return Integer.compare(a.idade, b.idade); // Ordena por idade crescente
            } else if (!a.cidade.equals(b.cidade)) {
                return a.cidade.compareTo(b.cidade); // Ordena por cidade em ordem alfabética
            } else {
                return a.nome.compareTo(b.nome); // Ordena por nome em ordem alfabética
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int q = scanner.nextInt();
        scanner.nextLine(); // Consumir o newline após o número

        while (q > 0) {
            List<Aluno> alunos = new ArrayList<>();

            // Leitura dos alunos
            for (int i = 0; i < q; i++) {
                String[] a = scanner.nextLine().split(" ");
                String nome = a[0];
                String cidade = a[1];
                int idade = Integer.parseInt(a[2]);
                Aluno aluno = new Aluno(nome, cidade, idade);
                alunos.add(aluno);
            }

            // Ordenação dos alunos usando o comparator AlunoComparator
            Collections.sort(alunos, new AlunoComparator());

            // Imprimir os nomes dos alunos ordenados
            for (Aluno aluno : alunos) {
                System.out.println(aluno.nome);
            }

            q = scanner.nextInt();
            scanner.nextLine(); // Consumir o newline após o número
        }

        scanner.close();
    }
}

File: Beecrowd_Parte5/Van_2693.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Estrutura que representa um aluno com nome, cidade e idade
typedef struct {
    char nome[21]; // 20 caracteres + 1 para o terminador null
    char cidade[21];
    int idade;
} Aluno;

// Função de comparação para qsort
int compararAluno(const void *a, const void *b) {
    Aluno *aluno1 = (Aluno *)a;
    Aluno *aluno2 = (Aluno *)b;

    if (aluno1->idade != aluno2->idade) {
        return aluno1->idade - aluno2->idade; // Ordena por idade crescente
    } else {
        int cmpCidade = strcmp(aluno1->cidade, aluno2->cidade);
        if (cmpCidade != 0) {
            return cmpCidade; // Ordena por cidade em ordem alfabética
        } else {
            return strcmp(aluno1->nome, aluno2->nome); // Ordena por nome em ordem alfabética
        }
    }
}

int main() {
    int q;
    while (scanf("%d", &q) == 1 && q > 0) {
        getchar(); // Consumir o newline após o número

        Aluno alunos[q];

        // Leitura dos alunos
        for (int i = 0; i < q; i++) {
            scanf("%s %s %d", alunos[i].nome, alunos[i].cidade, &alunos[i].idade);
            getchar(); // Consumir o newline após cada linha
        }

        // Ordenação dos alunos usando qsort com o comparador compararAluno
        qsort(alunos, q, sizeof(Aluno), compararAluno);

        // Imprimir os nomes dos alunos ordenados
        for (int i = 0; i < q; i++) {
            printf("%s\n", alunos[i].nome);
        }
    }

    return 0;
}
